### PR TITLE
[Rev 0 Demo] Customer Appointments Frontend

### LINKED ISSUE #294: [Rev 0 Demo] Customer Appointments Frontend
Design a page or component that looks nice

### LINKED ISSUE #312: [Rev 0 Demo] Customer Appointments Backend

### PR DESCRIPTION
**Pull request by TimChoy**
Close #294
Close #312

TODO:
- [x] Schedule by timeslot
- [ ] Create quote

### CODE REVIEW

File: src/pages/shop/[id]/index.tsx [hunk 1]
```diff
@@ -187,8 +187,11 @@ const Profile: NextPage = () => {
               {` ${shop.email}`}
             </span>
           </div>
-          <Button className={styles.shopProfileButtonGreen}>
-            Request Service
+          <Button
+            className={styles.shopProfileButtonGreen}
+            disabled={userType === null}
+          >
+            {userType === null ? "Login to Request Service" : "Request Service"}
```
Comment (LeonSo7) [inline src/pages/shop/[id]/index.tsx]: What you have is fine, another option is (up to you):

You can also let the customer click the button, and in the on click function you can open the login dialog (if the user is not logged in) using:
```
dispatch(setIsAuthDialogOpen( { isAuthDialogOpen: true; authDialogType: AuthDialogType.CUSTOMER });
```

File: src/pages/api/vehicle/customer/[id].ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { getServerAuthSession } from "@server/common/getServerAuthSession";
+import { getVehicleByCustomerId } from "@server/services/vehicleService";
+import { NextApiRequest, NextApiResponse } from "next";
+import type { Session } from "next-auth";
+
+const vehicleByCustomerIdHandler = async (
```
Comment (arkinmodi) [inline src/pages/api/vehicle/customer/[id].ts]: Put this file under `src/pages/api/customer/[id]/vehicle.ts`

File: src/pages/api/vehicle/customer/[id].ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    } else {
+      res.status(403).json({ message: "Forbidden." });
+    }
+  } else {
+    res.status(404).json({ message: "Vehicle not found." });
+  }
+};
+
+const isAuthorized = (session: Session) => {
+  return session.user.type === "CUSTOMER";
```
Comment (arkinmodi) [inline src/pages/api/vehicle/customer/[id].ts]: Should check that both the user is a customer and that the requester user id matches the customer id on the vehicle.

File: src/pages/api/shop/[id]/availabilities.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  if (typeof start !== "string" || typeof end !== "string") {
+    res.status(400).json({ message: "Invalid dates." });
+    return;
+  }
+
+  // const session = await getServerAuthSession({ req, res });
+  // if (!session) {
+  //   res.status(403).json({ message: "Forbidden." });
+  //   return;
+  // }
```
Comment (arkinmodi) [inline src/pages/api/shop/[id]/availabilities.ts]: Either uncomment this or delete this if we want to allow users who are not logged in to be able to view the availabilities.

File: src/pages/api/vehicle/customer/[id].ts (hunk 3 above)
Comment (LeonSo7) [reply src/pages/api/vehicle/customer/[id].ts]: Updated

**Review by arkinmodi — APPROVED** [review]
Backend looks good.

**Review by LeonSo7 — APPROVED** [review]

**Pull request merged**